example/model_usage.ipynb

```diff
@@ -2,132 +2,333 @@
+    "print(f\" Description:\") \n",
+    "pprint(model.description)\n",
+    "\n",
+    "print(\"\\n The authors of the model are: \")\n",
+    "pprint(model.authors)\n",
+    "\n",
+    "print(\"\\n It is maintained by:\")\n",
+    "pprint(model.maintainers)\n",
+    "\n",
+    "pprint(\"\\n License: \")\n",
+    "pprint(model.license)\n",
+    "\n",
+    "print(\"\\n If you use this model, you are expected to cite it as: \")\n",
+    "pprint(model.cite)\n",
+    "\n",
+    "print(f\"\\n Further documentation can be found here: {model.documentation.absolute()}\")\n",
+    "\n",
+    "if model.git_repo == None:\n",
+    "    print(f\"\\nNo associated GitHub repository.\")\n",
+    "else:\n",
+    "    print(f\"\\n GitHub repository: \")\n",
+    "    pprint(model.git_repo)\n",
+    "\n",
+    "print(f\"\\n Covers of the model '{model.name}' are: \")\n",
+    "for cover in model.covers:\n",
+    "    cover_data = imread(download(cover).path)\n",
+    "    plt.figure(figsize=(10, 10))\n",
+    "    plt.imshow(cover_data)\n",
+    "    plt.xticks([])\n",
+    "    plt.yticks([])\n",
+    "    plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.4 Get expected model inputs and outputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Each model expects Inputs and returns Outputs with specific features. The following code shows how to get the full metadata of the expected inputs and outputs of the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Model '{model.name}' requires {len(model.inputs)} input(s) with the following features:\")\n",
+    "for ipt in model.inputs:\n",
+    "    print(f\"\\ninput '{ipt.id}' with axes:\")\n",
+    "    pprint(ipt.axes)\n",
+    "    print(f\"Data description: {ipt.data}\")\n",
+    "    print(f\"Test tensor available at:  {ipt.test_tensor.source.absolute()}\")\n",
+    "    if len(ipt.preprocessing) > 1:\n",
+    "        print(\"This input is preprocessed with: \")\n",
+    "        for p in ipt.preprocessing:\n",
+    "            print(p)\n",
+    "\n",
+    "print(\"\\n-------------------------------------------------------------------------------\")\n",
+    "# # and what the model outputs are\n",
+    "print(f\"Model '{model.name}' requires {len(model.outputs)} output(s) with the following features:\")\n",
+    "for out in model.outputs:\n",
+    "    print(f\"\\noutput '{out.id}' with axes:\")\n",
+    "    pprint(out.axes)\n",
+    "    print(f\"Data description: {out.data}\")\n",
+    "    print(f\"Test tensor available at:  {out.test_tensor.source.absolute()}\")\n",
+    "    if len(out.postprocessing) > 1:\n",
+    "        print(\"This output is postprocessed with: \")\n",
+    "        for p in out.postprocessing:\n",
+    "            print(p)"
+   ]
```

Comment:
These parts are copied from https://github.com/bioimage-io/spec-bioimage-io/blob/main/example/load_model_and_create_your_own.ipynb
we should avoid this. If it is unclear that the model description can be inspected like that we can reference the spec example notebook here again

Reply:
The inspection of the model is indeed referenced to the spec tutorial. Maybe both `1.3 Inpsect model data`, and `1.4 Get expected model inputs and outputs` could be delegated to spec. For this tutorial, we shouldn't focus on the inspection, but we could have a quick summary of the model such as with `pprint(model)`? Would be nice maybe for spec to use `tabulate` package to have a nice table overview (e.g. authors, description, git repo, ...)

Reply:
yes, we should refer to the spec notebook instead of repeating so much.

An overview could be nice, for this purpose I also considered https://python-devtools.helpmanual.io/usage/#prettier-print , 
but I don't think we need to add that dependency to spec. 
We could add it [here](https://github.com/bioimage-io/core-bioimage-io-python/blob/3179f83cda85351f1ff4c00639580ffe9d0e32b6/setup.py#L51), but it does make this notebook a bit more convoluted. 

Maybe for a canonical rendering of a description we could simply render the YAML file, or use https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html#IPython.display.JSON to let jupyter display the metadata.
Simply printing `f"loaded {model.name}"` would do for now I suppose.

Reply:
yep actually the pprint, seems an easy solution :) Something like this should be fine
![image](https://github.com/user-attachments/assets/4d0dcc24-6c6f-4404-be66-3add757ad8c7)

Reply:
not sure what you mean...
- the pprint from devtools would handle pydantic models
- to avoid more dependencies we should probably just have a quick look at https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html#IPython.display.JSON , might be particularly suited for this situation (in a notebook)

we should not get hung up on this though. Especially anything not using another lib is totally fine without further consideration

Reply:
The above `pprint` is native python implementation, so no external dependencies. I just wanted to show, that it nicely renders the model's info. Not as pretty as it would be with pprint from devtools :)

For IPython.display.JSON couldn't make it work.